library/src/main/java/com/google/android/exoplayer/audio/AudioTrack.java

```diff
@@ -414,10 +412,15 @@ public int handleBuffer(ByteBuffer buffer, int offset, int size, long presentati
       bytesWritten = writeNonBlockingV21(audioTrack, buffer, temporaryBufferSize);
     }
 
-    temporaryBufferSize -= bytesWritten;
-    submittedBytes += bytesWritten;
-    if (temporaryBufferSize == 0) {
-      result |= RESULT_BUFFER_CONSUMED;
+    if (bytesWritten < 0) {
+        Log.w(TAG, "AudioTrack.write returned error code: " + bytesWritten);
```

Comment:
Yeah. I'll merge this and follow up with a change that properly propagates the error and fails playback.